src/components/Table/nestedCards.tsx

```diff
@@ -167,23 +177,33 @@ export function makeSpacer(height: number, openCards: NestedCardStyle[]) {
 export function maybeCreateChromeRow(
   columns: GridColumn<any>[],
   filteredRows: RowTuple<any>[],
-  chromeBuffer: JSX.Element[],
+  chromeBuffer: ChromeBuffer,
 ): void {
   if (chromeBuffer.length > 0) {
     filteredRows.push([
       undefined,
       // We add 2 to account for our dedicated open/close columns
-      <div css={Css.gc(`span ${columns.length + 2}`).$} data-chrome="true">
-        {chromeBuffer.map((c, i) => (
-          <Fragment key={i}>{c}</Fragment>
-        ))}
-      </div>,
+      <ChromeRow chromeBuffer={[...chromeBuffer]} columns={columns.length} />,
```

Comment:
Wow, I'm struggling to understand how this changed the perf, but it is nonetheless an amazing find...

So, what this does AFAICT is make `filteredRows` a homogeneous list of function components.

Previously it was `GridRow FC`, `JSX snippet`, `GridRow FC`, `JSX snippet`, like alternating (right?).

But, now it's all FC, like `GrowRow FC`, `Chrome FC`, `GridRow FC`, `Chrome FC`...

...and we're not even `React.memo`-ing the `ChromeRow`...

But, okay, I get here, we were still like doing a bunch of `Css.gc...` calls, and `React.createElement(div)`, and emotion `css=`, for ~600 chrome rows when we really only needed to render 10-20 of them.

Got it.

Really great find @chr1sjf0x !

Reply:
Thanks! You got the difference exactly. Defer the work until render time, so it's done only for the needed rows.